packages/node_modules/@webex/webex-core/src/lib/credentials/credentials.js

```diff
@@ -129,6 +129,25 @@ const Credentials = WebexPlugin.extend({
     /* eslint-enable camelcase */
   },
 
+  /**
+   * Extract the OrgId [realm] from a provided user token.
+   *
+   * @param {string} token - The user token to extract the OrgId from.
+   * @returns {string} - The OrgId.
+   */
+  extractOrgIdFromUserToken(token = '') {
+    // Split the provided token into subsections.
+    const fields = token.split('_');
+
+    // Validate that the provided token has the proper amount of sections.
+    if (fields.length !== 3) {
+      return undefined;
```

Comment:
This should throw when a token is invalid.

Reply:
Not in my implementation plans. See [this JIRA](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-125201) and [this comment](https://github.com/webex/webex-js-sdk/pull/1551#discussion_r387710737)

Reply:
This was discussed in a 1:1 chat. It was decided to leave this method as is for another implementation to consume.

Reply:
Making decisions over 1:1 chats and giving that as an answer doesn't feel satisfactory, could you explain the reasoning for making such decision?
I agree with @adamweeks that if something is not valid, it should throw an exception, and whatever consumes it should handle the exception how it sees fit

Reply:
Here's a larger example showing the logic needed to implement the planned subtask if we were to throw errors on each of the extraction methods and if we did not throw errors on each of the extraction methods:

```js
class credentials {
  /* ... */
  // Method extracs the OrgId from a JWT.
  // This could be marked private.
  extractOrgIdFromJWT(token = '') {}

  // Method extracs the OrgId from a user token.
  // This could be marked private.
  extractOrgIdFromUserToken(token = '') {}

  // Method gets the OrgId if determinable. Public method.
  determineOrgId() {
    const orgId = this.orgId ||
      this.extractOrgIdFromJWT(this.credentials.access_token) ||
      this.extractOrgIdFromUserToken(this.credentials.access_token) ||
      this.extractOrgIdFromUserToken(this.credentials.refresh_token);
      /** other ways to get the OrgID from credentials can be amended */

    if (!orgId) {
      throw new Error('excplicit error message');
    }

    return orgId;
  }

  determineOrgIdIfExtractorsThrowErrors() {
    if (this.orgId) {
      return this.orgId;
    }

    try {
      return this.extractOrgIdFromJWT(this.credentials.access_token);
    }
    catch {
      try {
        return this.extractOrgIdFromUserToken(this.credentials.access_token);
      }
      catch {
        try {
          return this.extractOrgIdFromUserToken(this.credentials.refresh_token);
        }
        catch {
          throw new Error('explicit error message');
        }
      }
    }
  }
  /* ... */
}
```

My biggest concern is the excessive nesting of these try/catch blocks. I may be mistaken, is there a better way to handle calling multiple error-throwing methods in the case that you are expecting them to throw a majority of the time?

Reply:
Here's a larger example showing the logic needed to implement the planned subtask if we were to throw errors on each of the extraction methods and if we did not throw errors on each of the extraction methods:

```js
class credentials {
  /* ... */
  // Method extracs the OrgId from a JWT.
  // This could be marked private.
  extractOrgIdFromJWT(token = '') {}

  // Method extracs the OrgId from a user token.
  // This could be marked private.
  extractOrgIdFromUserToken(token = '') {}

  // Method gets the OrgId if determinable. Public method.
  determineOrgId() {
    const orgId = this.orgId ||
      this.extractOrgIdFromJWT(this.credentials.access_token) ||
      this.extractOrgIdFromUserToken(this.credentials.access_token) ||
      this.extractOrgIdFromUserToken(this.credentials.refresh_token);
      /** other ways to get the OrgID from credentials can be amended */

    if (!orgId) {
      throw new Error('excplicit error message');
    }

    return orgId;
  }

  determineOrgIdIfExtractorsThrowErrors() {
    if (this.orgId) {
      return this.orgId;
    }

    try {
      return this.extractOrgIdFromJWT(this.credentials.access_token);
    }
    catch {
      try {
        return this.extractOrgIdFromUserToken(this.credentials.access_token);
      }
      catch {
        try {
          return this.extractOrgIdFromUserToken(this.credentials.refresh_token);
        }
        catch {
          throw new Error('explicit error message');
        }
      }
    }
  }
  /* ... */
}
```

My biggest concern is the excessive nesting of these try/catch blocks. I may be mistaken, is there a better way to handle calling multiple error-throwing methods in the case that you are expecting them to throw a majority of the time?

Reply:
Alternatively, this could just be linear try-catch statements since return is called in each:
```js
try {
  return this.extractOrgIdFromJWT(this.credentials.access_token);
} catch {} // do nothing
/* repeat */
```

Reply:
@InteractiveTimmy, you don't need to nest the `try-catch` for each type of token, they can all be under one. In case of failure, we can raise the error and let the developer know what went wrong, instead of failing silently.
Code would be something like

``js
determineOrgId() {
    const orgId = null;

    try {
      orgId = this.orgId ||
      this.extractOrgIdFromJWT(this.credentials.access_token) ||
      this.extractOrgIdFromUserToken(this.credentials.access_token) ||
      this.extractOrgIdFromUserToken(this.credentials.refresh_token);
   catch() {
       ...
   }

Reply:
In the above block, if `this.extractOrgIdFromJWT(this.credentials.access_token)` throws an error, does it continue to the next line: `this.extractOrgIdFromUserToken(this.credentials.access_token)` or continue to the `catch` statement?

Reply:
This is my suggestion 🤷🏻‍♀️, but do what aligns with the code design everywhere, or is more readable (although the latter can be a bit subjective)

Reply:
No, you're right, this wouldn't work in this case. Once it gets an error the control flow will go into the `catch`